src/components/contact-govuk-one-login/contact-govuk-one-login-controller.ts

```diff
@@ -12,18 +12,51 @@ export function contactGet(req: Request, res: Response): void {
       "Request to contact-govuk-one-login page did not contain a valid fromURL in the request or session"
     );
   }
+  // optional fields from mobile
+  const themeId = getValueFromRequestOrSession(req, "themeId");
+  const appSessionId = getValueFromRequestOrSession(req, "appSessionId");
+  const appErrorCode = getValueFromRequestOrSession(req, "appErrorCode");
+
   const data = {
     fromURL: fromURL,
+    appSessionId: appSessionId,
```

Comment:
Same here, can be simplified to
`const data = {
    fromURL,
    appSessionId,
    appErrorCode,
    themeId
  };`

Reply:
done